packages/astro/src/core/build/generate.ts

```diff
@@ -175,13 +175,12 @@ async function generatePath(
 	}
 
 	debug('build', `Generating: ${pathname}`);
-
-	// If a base path was provided, append it to the site URL. This ensures that
+	
+	// If a base path was provided, append it to the stie URL. This ensures that
 	// all injected scripts and links are referenced relative to the site and subpath.
-	const site =
-		astroConfig.base !== '/'
-			? joinPaths(astroConfig.site?.toString() || 'http://localhost/', astroConfig.base)
-			: astroConfig.site;
+	const site = astroConfig.base !== '/'
```

Comment:
This makes sure that the subpath is always used for scripts and styles that Astro injects into the page